### ISSUE TITLE
[Bug Report]: cannot subscribe to systemtopic

### ISSUE THREAD

**Issue by davidkarlsen** [Azure/ResourceModules#2881]
### Describe the bug

In https://github.com/Azure/ResourceModules/blob/main/modules/Microsoft.EventGrid/eventSubscriptions/deploy.bicep it is assumed that the topic is a `Microsoft.EventGrid/topics@2022-06-15` but very often this might be a systemTopic: https://learn.microsoft.com/en-us/azure/templates/microsoft.eventgrid/systemtopics?pivots=deployment-language-bicep

parameters should be taken for both types, and use whichever is defined.

Either that, or take in resourceIds

### To reproduce

```


module eventGrid 'br/dnb:bicep/modules/microsoft.eventgrid.systemtopics:0.4.1213' = {
  name: '${fullResourceName}-eg'
  params: {
    name: '${fullResourceName}-eg'
    source: storageAccount.outputs.resourceId
    topicType: 'Microsoft.Storage.StorageAccounts'
    diagnosticLogsRetentionInDays: diagnosticLogsRetentionInDays
    diagnosticWorkspaceId: logAnalyticsWorkspace.id
    systemAssignedIdentity: true
    tags: commonTags
  }
}

module mftBgcUploadSubscription 'br/dnb:bicep/modules/microsoft.eventgrid.eventsubscriptions:0.1.1209' = {
  name: 'uploadSub'
  params: {
    destination: {
      endpointType: 'StorageQueue'
      properties: {
        queueMessageTimeToLiveInSeconds: 60*60*24 // one day
        queueName: queueName
        resourceId: storageAccount.outputs.resourceId
      }
    }
    eventDeliverySchema: 'CloudEventSchemaV1_0'
    name: eventGrid.name
    eventGridTopicName: 'pab-ei-dev-eg'
    retryPolicy: {
      //TODO
    }
  }
}

```

it will fail with:
The Resource 'Microsoft.EventGrid/topics/pab-ei-dev-eg' under resource group 'XXX' was not found.

because the path is Microsoft.EventGrid/systemtopics/pab-ei-dev-eg

**Comment by davidkarlsen** [comment]
Any progress?

**Comment by rahalan** [comment]
@davidkarlsen Sorry for the delay. Should be fixed with #3107

**Comment by davidkarlsen** [comment]
> @davidkarlsen Sorry for the delay. Should be fixed with #3107

But that is still not merged.

**Comment by davidkarlsen** [comment]
Closing since https://github.com/Azure/ResourceModules/pull/3281 was merged. Will reopen if issues.
